### ISSUE TITLE
[Question] Post processing many telemetries from one or multiple devices in a dashboard (e.g. add/subtract/divide/multiply/average)

### ISSUE THREAD

**Issue by Civilduino** [thingsboard/thingsboard#8886]
**Component**

 * UI, Dashboard
 * Generic

**Description**
This is a continuation of this question #7153.

Sometimes, it is necessary to do some post-processing for a given data key or multiple data keys at the same time. For a single data key, post-processing is easily done using the "Data key configuration"  under the datasources inside the widget (see image below).

![image](https://github.com/thingsboard/thingsboard/assets/27728926/4bd68160-9540-47bd-bc9f-ee7d4055a4f5)

But how do you do the same if you have multiple data keys in the same Entity (device), for example, the average of Frequency0, Frequency1 and Frequency2?

![image](https://github.com/thingsboard/thingsboard/assets/27728926/e7c294f6-c31a-42ac-8ed7-1dcbff10e653)

or multiple Entities (devices) with a single data key, for example, add Frequency0 from Sensor1 and Sensor2?

![image](https://github.com/thingsboard/thingsboard/assets/27728926/2a0d59c7-d856-4d17-a7e0-61e69b668590)

In summary, how do you post-process several devices and multiple telemetries in the dashboard? In this case, there is no need to store the data in the database using a RuleChain.

**Comment by santobedi** [comment]
@ashvayka @Civilduino Any updates on this issue?

**Comment by dm9de** [comment]
TB's default widgets don't provide any way to combine data from different data sources. I suspect this is because thingsboard can't guarantee that data will always be available or that the data sources will have a 1-to-1 correspondence with each other (so e.g. what should TB do if a value for `Frequency0` was available, but `Frequency1` was not, or if `Frequency0` and `Frequency1` values have different time stamps?).

As far as I can tell, there are two options:

- do the processing in a rule chain (potentially using the 'originator enrichment' node to retrieve the relevant telemetry points, if they are sent in different messages); in this case, you'll need to store the result as a new telemetry value against the entity.
- create a new widget type which includes the logic for processing the data, and handling any data syncing / data availability issues that come up

**Comment by Aryan2892** [comment]
Can anyone provide me with the js for the this kind of a widget. I desperately need it and chatgpt is just providing me with worthless code.
Thanks in advance.
